ni/niveristand-communications-bus-template
# Cloning - support PPLs renamed, rebuilt, and replaced in project.

- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/niveristand-communications-bus-template/blob/main/CONTRIBUTING.md).

### What does this Pull Request accomplish?

The cloning process now builds the PPLs so they initially exist. The names are then changed in the support project, and the PPLs are built again. The newly named PPLs are replaced in the project. Finally, the originally built PPLs are deleted from the Includes folder.

Edited VIs:

- `Clone Template Main.vi` - VIs added
- `Copy Template Source to New Location.vi` now initializes globals.
- `Edit Paths in XML File.vi` uses new Find and Replace subVI.
- `Edit Project File Build Spec Destinations.vi` - uses new Find and Replace subVI.
- `Edit VI Constants.vi` - cleared defaults.
- `Remove Test Folder From lvproj Temporarily.vi` now uses **CD lvproj name** global.
- `Replace VI and XML GUIDs.vi` - cleared defaults.
- `Template Tool Global.vi` - added two items.
- `Update CD Name in XML File.vi` now uses xml name global.

New VIs:

- `Build Support lvproj PPLs.vi`
- `Includes Folder Auto-Populating On or Off.vi` - have to turn off auto-populating in order for replace to work.
- `Edit Support lvproj PPL Names.vi`
- `Replace Project PPLs.vi`
- `Delete Old PPLs.vi`
- `Find and Replace in Text File.vi` - new subVI.

### Why should this Pull Request be merged?

This covers part of the third "Unique library names" user story for cloning the Comms Bus Template.

### What testing has been done?

Testing on my dev VM includes:

- Run `Clone Template Main.vi`  to clone the source.
- Confirm the newly built and named PPLs are in the project.
- Build the two CD build specifications by hand: `Configuration Release` and `Engine Release`.
- Confirm files get copied to _C:\Users\Public\Documents\National Instruments\NI VeriStand 2020\Custom Devices_.
- Confirm the CD shows in System Explorer under  **Custom Devices>National Instruments** and appears to function.
